packages/cli/src/commands/multisig/approve.ts

```diff
@@ -0,0 +1,49 @@
+import { flags } from '@oclif/command'
+import { BaseCommand } from '../../base'
+import { newCheckBuilder } from '../../utils/checks'
+import { displaySendTx } from '../../utils/cli'
+import { Flags } from '../../utils/command'
+
+export default class ApproveMultiSig extends BaseCommand {
+  static description = 'Approves an existing transaction on a multi-sig contract'
+
+  static flags = {
+    ...BaseCommand.flags,
+    from: Flags.address({
+      required: true,
+      description: 'Account approving the multi-sig transaction',
+    }),
+    for: Flags.address({
+      required: true,
+      description: 'Address of the multi-sig contract',
+    }),
+    tx: flags.integer({
+      required: true,
+      description: 'Transaction to approve',
+    }),
+  }
+
+  static examples = [
+    'approve --from 0x6ecbe1db9ef729cbe972c83fb886247691fb6beb --for 0x5409ed021d9299bf6814279a6a1411a7e866a631 --tx 3',
+  ]
+
+  async run() {
+    const res = this.parse(ApproveMultiSig)
+    const account = res.flags.from
+    this.kit.defaultAccount = account
+
+    const multisig = await this.kit.contracts.getMultiSig(res.flags.for)
+
+    const checkBuilder = newCheckBuilder(this)
+      .isMultiSigOwner(account, multisig)
+      .addCheck(`${res.flags.tx} is existing transaction`, async () => {
```

Comment:
While we're waiting for primitive tests to pass, here is one of the small improvements I suggested. 

There is 1 pending transaction:

```sh
$ bin/run multisig:show 0x37A6bA9CFC3C4C7f82bDf1Fe1B7cEb410E78738E

Owners: 
  0: 0x303C22e6ef01CbA9d03259248863836CB91336D5
  1: 0xEb08a36D153827CC750dd264c9e7360A3174Dc27
Required confirmations: 2 
Required confirmations (internal): 2 
Transactions: 1
```

So passing `--tx 1` is incorrect because transactions are zero indexed, `--tx 0` is correct. But, the error message incorrectly states `✘  1 is existing transaction`, when it should probably say `✘  1 is not an existing transaction`.

```sh
$ bin/run multisig:approve --from 0xEb08a36D153827CC750dd264c9e7360A3174Dc27 --for 0x37A6bA9CFC3C4C7f82bDf1Fe1B7cEb410E78738E > --tx 1 --privateKey <privateKey>

Running Checks:
  ✔  The provided address is an owner of the multisig 
  ✘  1 is existing transaction 
›   Error: Some checks didn't pass!
```

```suggestion
      .addCheck(`${res.flags.tx} is not an existing transaction`, async () => {
```

Reply:
this would be wrong. because we WANT it to be an existing transaction. 

do what you want "show WHAT the error is when it failes" we will need to rewrite how addCheck works to take both a happy and sad messege

Reply:
Ah, I see 👍 Looks like `addCheck` takes an optional error message:

```ts
addCheck(name: string, predicate: () => boolean | Promise<boolean>, errorMessage?: string | undefined)
```

I tried giving it an error message here for example: 

```diff
const checkBuilder = newCheckBuilder(this)
      .isMultiSigOwner(account, multisig)
      .addCheck(
        `${res.flags.tx} is existing transaction`,
        async () => {
          const max = await multisig.getTransactionCount(true, true)
          return res.flags.tx < max
        },
+        `${res.flags.tx} is not an existing transaction`
      )
```

But the CLI output is suboptimal: 

```sh
bin/run multisig:approve --from 0xEb08a36D153827CC750dd264c9e7360A3174Dc27 --for 0x37A6bA9CFC3C4C7f82bDf1Fe1B7cEb410E78738E --tx 10 --privateKey <privatekey>

Running Checks:
   ✔  The provided address is an owner of the multisig 
   ✘  10 is existing transaction 10 is not an existing transaction
 ›   Error: Some checks didn't pass!
```

Reply:
How about this? 

```diff
     const checkBuilder = newCheckBuilder(this)
       .isMultiSigOwner(account, multisig)
-      .addCheck(`${res.flags.tx} is existing transaction`, async () => {
-        const max = await multisig.getTransactionCount(true, true)
-        return res.flags.tx < max
-      })
+      .addCheck(
+        `Checking that ${res.flags.tx} is an existing transaction.`,
+        async () => {
+          const max = await multisig.getTransactionCount(true, true)
+          return res.flags.tx < max
+        },
+        `${res.flags.tx} is not an existing transaction.`
+      )
```

Gives the following output? 

Error: 

```sh
bin/run multisig:approve --from 0xEb08a36D153827CC750dd264c9e7360A3174Dc27 --for 0x37A6bA9CFC3C4C7f82bDf1Fe1B7cEb410E78738E --tx 10 --privateKey <privateKey>

Running Checks:
   ✔  The provided address is an owner of the multisig 
   ✘  Checking that 10 is an existing transaction. 10 is not an existing transaction.
 ›   Error: Some checks didn't pass!
```

Success:

```sh
bin/run multisig:approve --from 0xEb08a36D153827CC750dd264c9e7360A3174Dc27 --for 0x37A6bA9CFC3C4C7f82bDf1Fe1B7cEb410E78738E --tx 10 --privateKey <privateKey>

Running Checks:
   ✔  The provided address is an owner of the multisig 
   ✔  Checking that 0 is an existing transaction. 
All checks passed
```

Reply:
`Checking that 10 is an existing transaction. (No transaction with ID 10)`

`Checking that 10 is an existing transaction. (Failed: no transaction found)`

`Checking that 10 is an existing transaction. (It is not)`

`Checking that 10 is an existing transaction. But could not find it`

Reply:
some ideas if you like

Reply:
Nice, good suggestions. I chose the below.

Error:

```sh
Running Checks:
   ✔  The provided address is an owner of the multisig 
   ✘  Checking that 10 is an existing transaction. (Failed: No transaction with index 10 found)
 ›   Error: Some checks didn't pass!
```

Success:

```sh
Running Checks:
   ✔  The provided address is an owner of the multisig 
   ✔  Checking that 0 is an existing transaction. 
All checks passed
```

If you're happy with that, I'll make a small commit, run the tests again (😿), and merge it, so this can be released with the other CLI changes: 

- https://github.com/celo-org/celo-monorepo/pull/10699

Reply:
Added in this commit https://github.com/celo-org/celo-monorepo/pull/10693/commits/54c0feca773442b44a65c03e2742f675ea0ed70a